### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by jasozh**
## Summary

Update dependency versions in `package.json`. Did some preliminary testing to make sure nothing was broken.

### CODE REVIEW

**Review by akinfelami — COMMENTED** [review]
Can we leave this open for now ? until we merge the other large PR. I don't assume anything will be broken, but for good measure, I say we merge this after testing with that PR

**Comment by jasozh** [comment]
I think it should be fine because we can always revert back to the old yarn.lock and package.json versions if an issue does come up. I wasn't able to find any issues but feel free to test it whenever you're free. I think reading through the changelog might be a bit too much, but I understand the concern.

File: frontend/package.json [hunk 1]
```diff
@@ -11,36 +11,35 @@
...
-    "firebase": "^9.18.0",
-    "next": "13.1.6",
-    "react": "18.2.0",
-    "react-dom": "18.2.0",
+    "date-fns": "^3.3.1",
+    "dayjs": "^1.11.10",
+    "eslint": "^8.56.0",
+    "eslint-config-next": "^14.1.0",
+    "firebase": "^10.8.0",
+    "next": "^14.1.0",
```
Comment (akinfelami) [inline frontend/package.json:31]: https://nextjs.org/blog/next-14#other-changes

**Review by akinfelami — APPROVED** [review]

**Review by akinfelami — CHANGES_REQUESTED** [review]
hold off to fix  merge conflicts

**Review by akinfelami — APPROVED** [review]
good to go!

**Pull request merged**